### ISSUE TITLE
[MIUI] Unable to show certain Indian scripts

### ISSUE THREAD

**Issue by bernardkkt** [nongthaihoang/unicode_google_sans_font#6]
First of all, thank you @nongthaihoang for creating this project!

Here's the glitch that I have found on my phone running MIUI 11 (from xiaomi.eu): when viewing websites that contain certain Indian scripts, those scripts are rendered as boxes with a cross in the middle of each box. It can be reproduced with the following steps:
1. Download the TWRP version of the ZIP file.
2. Reboot the phone to TWRP.
3. Flash the downloaded ZIP file.
4. Reboot the phone to normal mode.
5. Open https://hi.wikipedia.org with Google Chrome.

It is also possible to tell which scripts are unable to display properly by visiting https://m.facebook.com/language.php, which contains a list of all supported languages by Facebook. For this instance, some Indian languages in their native scripts are not shown in the list.

Hopefully we will get an update for the fix :) Thank you!

**Comment by nongthaihoang** [comment]
Hello. Thank you for reporting.
I need few things:
- Screenshots of the websites before and after installing this font.
- The options you chose when installing the font.
- Does it happen anywhere else besides Google Chrome?
- Send me your original /system/etc/fonts.xml file.
- Any differences with choosing Yes to the last question?

Waiting for you.

**Comment by bernardkkt** [comment]
Hello! Sure, see below :)

   - Screenshots:
      - Before
      [image: before.jpg]
      - After
      [image: after.jpg]
      - Q&A
      - WHERE to install? -- 1. Full
      - Which STYLE? -- 1. Default
      - ROM? -- 3. MIUI 11
      - Use Android default font reference? -- No
      - The problem also occurs on Google Keep. In addition, the language
   menu inside MIUI's system settings fails to show Hindi as one of the
   selections.
   - The content of the fonts.xml file: https://pastebin.com/TvP9vCN4
   - I tried both options, but it did not make any difference.

Thank you!

**Comment by bernardkkt** [comment]
Hello! Sure, see below :)
   - Screenshots:
      - Before
      [image: https://imgur.com/a/KPiL2Ar]
      - After
      [image: https://imgur.com/a/dGO3323]
   - Q&A
      - WHERE to install? -- 1. Full
      - Which STYLE? -- 1. Default
      - ROM? -- 3. MIUI 11
      - Use Android default font reference? -- No
   - The problem also occurs on Google Keep. In addition, the language menu inside MIUI's system settings fails to show Hindi as one of the selections.
   - The content of the fonts.xml file: https://pastebin.com/TvP9vCN4
   - I tried both options, but it did not make any difference.

Thank you!

**Comment by nongthaihoang** [comment]
If even using Android default font reference (font.xml) can't solve the issue. The chance to fix it is very small ☹️ Unless I have MIUI on my phone.
Do you have any custom font that works properly?

**Comment by bernardkkt** [comment]
No, I do not have any custom font other than yours (San Francisco,
Google Sans). I am reluctant to root my device, as well as using the
official MIUI theme application for applying new fonts as it requires
an account. Let us see if there are other MIUI users know how to fix
it. :D
Thank you once again!

**Comment by nongthaihoang** [comment]
You said the issue happens on Google Chrome and Keep. So it is not a system wide issue, isn't it? What about other browser such as Firefox, Samsung Internet?

**Comment by bernardkkt** [comment]
I just downloaded Firefox, installed the font again, and opened the website
with Firefox. The page could be loaded correctly!
But I feel it should still be a system wide issue, because the glitch does
occur at the language menu inside MIUI's system settings.

>

**Comment by nongthaihoang** [comment]
The issue need to be fixed from inside MIUI. The system can't find Hindi font.
- What is the default font then? Roboto or MiLanPro?
- Can you send me a zip that includes all fonts in /system/fonts?

**Comment by bernardkkt** [comment]
- It is Roboto.
- Here you go https://takeafile.com/?f=gecusawara. Pardon me that I could only provide it as a tar.gz archive. Not sure why my file explorer refuses to compress it into ZIP.

**Comment by nongthaihoang** [comment]
The link you send me is not available.
If the default font is Roboto, then I see no reason why your language does not show.
You can try these each time and tell me the result.
- When installing, do not choose MIUI, choose the default one AOSP/LOS instead.
- Manually replace the file /system/fonts/Roboto-Regular.ttf with the one in the font zip (ie custom/bf/Roboto-Regular.ttf).

**Comment by bernardkkt** [comment]
Sorry, didn't realise the link is invalid. I have re-uploaded the file to another file hosting service (https://framadrop.org/r/J1OHZTO04v#XEjwCv+q18DPjhyun/53cPERfELBXDGckmC3YtD6Vi8=).
- After choosing AOSP/LOS, the font does not seem to be applied at the system UI, but when browsing Reddit with Google Chrome, the fonts appeared as Google Sans in the webpage. I was also able to view Wikipedia in Hindi correctly.
- Then I proceed to replacing "Roboto-Regular.ttf" at TWRP as suggested. Nothing changed.

I also installed it again by choosing MIUI and replaced "Roboto-Regular.ttf". It did not help either, it was the same glitch reported in the first post.
